WordPress/gutenberg
# Site Editor: Add new Comments Query Loop block

## Description
Fixes https://github.com/WordPress/gutenberg/issues/34996
Fixes https://github.com/WordPress/gutenberg/issues/35000

- This PR is adding a new Comments Query block to `block-library`.
- It includes only the bare skeleton of the block without all the attributes that should be present in the final version.
- I've included a stub implementation of the [`core/comment-template` block](https://github.com/WordPress/gutenberg/issues/35000) which is kept to the bare minimum to make the comments query work.

## How has this been tested?
Tested manually:

1. Create a post
2. Type `/comment`
3. Select the `Comment Query Loop` block
4. Change the `perPage`, `pages` or `offset` attributes in the toolbar and observe them change in the UI

### Environment details
- WordPress Version: 5.9-alpha-51831
- Database: `wp-env` default
- PHP Version: `wp-env` default
- Node: `14.17.6`
- npm: `6.14.15`

## Screenshots
<img width="600" alt="Screen Shot 2021-10-07 at 20 09 01" src="https://user-images.githubusercontent.com/5417266/136482656-650c5a47-59a8-4d72-94ad-e7f5fbb55b3a.png">
<img width="600" alt="Screen Shot 2021-10-07 at 20 09 35" src="https://user-images.githubusercontent.com/5417266/136482697-fba25b2e-0b0a-4e53-9b78-be83e98fbf1a.png">
<img width="600" alt="Screen Shot 2021-10-07 at 20 10 57" src="https://user-images.githubusercontent.com/5417266/136482791-45bb94d9-4795-4c4d-bd3f-85a0133e38de.png">
<img width="619" alt="Screen Shot 2021-10-07 at 20 13 45" src="https://user-images.githubusercontent.com/5417266/136482972-8a6f2f96-faa3-440e-829f-4d1fdbe3e949.png">

## Types of changes
- Type: Experimental
- Needs Testing
- New feature (Block Editor)

## Checklist:
